### PR TITLE
Remove unneeded registry creation

### PR DESCRIPTION
**Pull request by Bakual**
Pull Request for similar Issue #12557.
### Summary of Changes

Referenced PR changes one instance where we tried to create a Registry object from a Registry object.
After scanning the PR which introduced that bug I found several more instances of the same error. This is corrected in this PR.
### Testing Instructions

I did this PR based on code review and debugging (looking at the values of he params during runtime).
To test you can do the same review/debugging: If the params property is already a Registry object it will fail to create one and it doesn't make sense to create one for obvious reasons.

To test in real you need to set some parameters at item level and check that the item params take precedence over menu/component params if needed.
- Contact Item in category, featured and single contact view
- Newsfeed Item in category and single newsfeed view
- In the search view for the traditional search (com_search) you need to check the behavior of the page title parameter in the menu item. If none is specified it should fall back to the default language string, if one is specified it has to take the specified one.
- Last change is in the template override feature, but I'm not 100% sure how to test it. Adding `?template=beez3` to the URL brings you to the code I changed, however I'm not sure which params are used there since no template style is specified. I guess it will just use default values but not sure.
  The code itself is a no brainer since a few lines above the params are already converted to Registry, so this can be reviewed easily.
### Documentation Changes Required

None

### CODE REVIEW

**Comment by laoneo** [comment]
Branch has a conflict, can you please fix it that I can test.

**Comment by Bakual** [comment]
Rebased

File: components/com_contact/views/category/view.html.php [hunk 1]
```diff
@@ -60,9 +60,8 @@ public function display($tpl = null)
 		foreach ($this->items as $item)
 		{
 			$item->slug = $item->alias ? ($item->id . ':' . $item->alias) : $item->id;
-			$temp       = new Registry($item->params);
 			$item->params = clone $this->params;
-			$item->params->merge($temp);
+			$item->params->merge($item->params);
```
Comment (laoneo) [inline components/com_contact/views/category/view.html.php]: You are merging here two time $this->params but $item->params get lost on the clone. This applies to all view.

**Comment by laoneo** [comment]
A bit hard to set up a test environment. Can you tell us how we can reproduce the issue.

File: components/com_contact/views/category/view.html.php (hunk 1 above)
Comment (Bakual) [inline components/com_contact/views/category/view.html.php]: Now that you say it, this looks indeed wrong to me as well.

**Comment by Bakual** [comment]
I don't remember if there was an issue in those cases.
It's just that `$item->params` is already a Registry object and doesn't need to be created again with `new Registry($item->params)`. Trying to do that actually results in an error and the merging doesn't work correctly. The merge will end up with a Registry object within a Registry object instead of merging it.
Easy to see when you debug it, but I haven't looked where exactly those come from and how the issue could be made viewable.

**Comment by Bakual** [comment]
Should be fine now.

**Comment by dgt41** [comment]
I have tested this item :white_check_mark: successfully on 0236a98de1dcca97f3e1b339fd9bf3aa1d6589f5<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12571">issues.joomla.org/tracker/joomla-cms/12571</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 0236a98de1dcca97f3e1b339fd9bf3aa1d6589f5<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12571">issues.joomla.org/tracker/joomla-cms/12571</a>.</sub>

**Comment by jeckodevelopment** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12571">issues.joomla.org/tracker/joomla-cms/12571</a>.</sub>

**Pull request merged**